src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -115,21 +88,21 @@ update_shell(unsigned int index, const std::string& flashType,
   if(!flasher.isValid())
     throw xrt_core::error(boost::str(boost::format("%d is an invalid index") % index));
 
-  if (primary.empty())
+  if (image_paths.empty())
     throw xrt_core::error("Base not specified");
 
-  auto pri = std::make_unique<firmwareImage>(primary.c_str(), MCS_FIRMWARE_PRIMARY);
+  auto pri = std::make_unique<firmwareImage>(image_paths[0].c_str(), MCS_FIRMWARE_PRIMARY);
```

Comment:
I used [0] instead of front to match the other reference in the function using [1] on [https://github.com/Xilinx/XRT/pull/6095/files#diff-5b1fd3c2284500459e3cda9231fc218d31e9798c321e9ea84e9ecc79e054cef0R100]()

Reply:
Both styles look good.